### PR TITLE
Migration/ornament

### PR DESCRIPTION
**Pull request by rvilarl**
Ornament!

### CODE REVIEW

**Comment by rvilarl** [comment]
Rebase done!

File: src/ornament.ts [hunk 1]
```diff
@@ -245,7 +273,7 @@ export class Ornament extends Modifier {
     let lineSpacing = 1;
 
     // Beamed stems are longer than quarter note stems, adjust accordingly
-    if (!isPlacedOnNoteheadSide && note.beam) {
+    if (!isPlacedOnNoteheadSide && note.getBeam()) {
```
Comment (0xfe) [inline src/ornament.ts:276]: Don't we have a `hasBeam` which is better?

File: src/ornament.ts [hunk 2]
```diff
@@ -270,12 +298,12 @@ export class Ornament extends Modifier {
       if (this.delayXShift !== undefined) {
         delayXShift = this.delayXShift;
       } else {
-        delayXShift += this.glyph.getMetrics().width / 2;
+        delayXShift += (this.glyph.getMetrics().width ?? 0) / 2;
```
Comment (0xfe) [inline src/ornament.ts:301]: This is a silent failure, check for `width` explicitly.

File: src/ornament.ts [hunk 3]
```diff
@@ -286,21 +314,21 @@ export class Ornament extends Modifier {
 
     if (this.accidentalLower) {
       this.accidentalLower.render(ctx, glyphX, glyphY);
-      glyphY -= this.accidentalLower.getMetrics().height;
+      glyphY -= this.accidentalLower.getMetrics().height ?? 0;
```
Comment (0xfe) [inline src/ornament.ts:317]: Same, silent failure here.

File: src/ornament.ts [hunk 4]
```diff
@@ -286,21 +314,21 @@ export class Ornament extends Modifier {
...
     if (note.getLineNumber() < 5 && Ornament.ornamentNoteTransition.indexOf(this.type) >= 0) {
-      glyphY = note.getStave().getBoundingBox().y + 40;
+      glyphY = note.checkStave().getBoundingBox().getY() + 40;
     }
 
     this.glyph.render(ctx, glyphX + this.x_shift, glyphY);
 
     if (this.accidentalUpper) {
-      glyphY -= this.glyph.getMetrics().height + this.render_options.accidentalUpperPadding;
+      glyphY -= (this.glyph.getMetrics().height ?? 0) + this.render_options.accidentalUpperPadding;
```
Comment (0xfe) [inline src/ornament.ts:331]: Silent failure.

File: src/ornament.ts (hunk 1 above)
Comment (rvilarl) [reply src/ornament.ts]: done

File: src/ornament.ts (hunk 3 above)
Comment (rvilarl) [reply src/ornament.ts]: Resolved in glyph calculating height and width

File: src/ornament.ts (hunk 4 above)
Comment (rvilarl) [reply src/ornament.ts]: Done

**Comment by rvilarl** [comment]
Ready to merge!

File: src/ornament.ts (hunk 2 above)
Comment (rvilarl) [reply src/ornament.ts]: resolved

**Comment by 0xfe** [comment]
looks good

**Pull request merged**